src/components/item-icon.tsx

```diff
@@ -90,13 +92,15 @@ export const ItemIcon = React.memo(
     type: 'tx' | 'microblock' | 'block' | 'principal';
     tx?: Transaction | MempoolTransaction;
   } & GridProps) => {
+    const txStatus = useTransactionStatus(tx);
```

Comment:
Here, where we're using the fn, we may indeed wish to prevent multiple renders of the fn, but we can do this by memoizing the function.

```suggestion
    const txStatus = useMemo(() => getTransactionStatus(tx), [tx])
```